packages/sql-json-schema-mapper/index.js

```diff
@@ -87,4 +94,5 @@ function mapSQLEntityToJSONSchema (entity) {
   }
 }
 
-module.exports = { mapSQLTypeToOpenAPIType, mapSQLEntityToJSONSchema }
+const noRequiredSuffix = 'NoRequired'
```

Comment:
I'm worried about possible name collisions with users' schemas. What you can do is create a schema with the `definitions` keyword.

```js
{
  $id: 'Page',
  definitions: {
    origin: {/* original schema */},
    partial: {/* schema without requited schema */},
  }
}
```

and then use $refs like `Page#definition/origin` and `Page#definition/partial`

Reply:
Where `origin` schema definition will look like that
```js
{
  allOf: [
    { $ref: '#definitions/partial' },
    { required: [/* list of required fields */] }
  ]
}
```

Reply:
@marcopiraccini @iifawzi WDYT?

Reply:
I think this would be much better

Reply:
Yes, this will avoid the conflict name risk on schema. 
@iifawzi can you please update your PR with this suggestion?
This imply that we need to change also the TS generation, right @ivan-tymoshenko ?

Reply:
Sure I will work on updating the API.

Reply:
Why such schema ends up empty object in docs? am I missing something?
```JS
 app.addSchema({
  $id: 'Page',
  definitions: {
    partial: {
      title: 'test',
      description: `A Page`,
      type: 'object',
    },
  } 
})
```

Reply:
I guess because it has only `definitions` blocks and never use it.

Reply:
Everything is working as expected on the validation side after making the changes, only the swagger part is complaining it couldn't resolve the schemas. 
```
Could not resolve reference: Could not resolve pointer: /components/schemas/def-0/definitions/partial does not exist in document
```
```JS
  app.addSchema({
    $id: entity.name,
    type: 'object',
    definitions: {
      partial: {
        type: 'object',
        title: entity.name,
        description: `A ${entity.name}`,
        type: 'object',
        properties,
      },
      origin: {
        allOf: [
          { $ref: entity.name + '#/definitions/partial' },
          { required }
        ]
      }
    }
  });
```

```JS
  const entitySchemaNoRequired = {
    $ref: entity.name + '#/definitions/partial'
  }
  app.get('/', {
    schema: {
      response: {
        200: {
          type: 'array',
          items: entitySchemaNoRequired
        }
      }
    }
  }, async function (request, reply) { });

```

Reply:
There is a bug in @fastify/swagger https://github.com/fastify/fastify-swagger/issues/639

Reply:
Will https://github.com/fastify/fastify-swagger/pull/676 be enough?

Reply:
Yes.